debian/control

```diff
@@ -5,7 +5,7 @@ Build-Depends: debhelper (>= 10),
                cmake,
                libboost-dev,
                libevent-dev,
-               libfabric-dev,
+               libfabric-dev (= 1.14.0-1),
```

Comment:
Unfortunately we cannot do that for now until HPE upgrades their libfabric version. I definitely understand and I don't like to do that either but we need to be explicitly compatible with 1.14 for now to be able to install rpms on HPE systems and use CXI or we'd need to generate rpms directly on those systems, but that's not how things are currently done.

Reply:
I'm not sure I understand how `= 1.14.0-1` works but `>= 1.14.0-1` does not.  They both achieve the same minimum version.  Any kind of site-specific maximum version should be handled by the site admins, specifically installing the particular libfabric their site needs.

What version of libfabric is this HPE libfabric?

Reply:
It's based on 1.14.0. There is currently a conflict in the headers if the rpms use 1.15.0 to build mercury because it conflicts with the OPX provider which takes the same enum value as CXI. They have fixed the conflict in libfabric master but until they upgrade to a new libfabric (1.15.2 I believe will have the fix) we need to stick with 1.14.0 for the build (to be able to use those same rpms on those sites).

Reply:
To make sure I understand, if we build mercury with libfabric 1.15.0 that mercury RPM cannot be used on systems with libfabric 1.14.0 which these HPE systems are currently stuck on?  And they want to use the mercury RPMs that we build.  Is my understanding correct?

If so, then we upgraded to libfabric 1.15 too quickly and we should simply withdraw that upgrade until everyone can use libfabric 1.15.$something?  That way, we can continue to use `libfabric-dev (>= 1.14.0-1)` knowing that our build systems won't have access to any libfabric newer than 1.14.0-1, until we can update to something that does not have these problems that 1.15.0 has.

How long is it expected that we will need to be `= 1.14.0-1`?  We really cannot release a mercury with a daos GA release with such a requirement.  So if it's just something we need to do internally for a short while (i.e. will be resolved before we release any new mercury RPM (which would be in any of our 2.{0,2,4}.x upcoming releases) then we could probably live with it for that short while.

Reply:
Correct, and that's because HPE have their own flavor of libfabric currently based on 1.14.0 but with additional changes that are specific to supporting their CXI provider, they even provide an additional header `fi_cxi_ext.h` for extensions to libfabric (we should consider how to handle that in the future if we ever start using those extensions). In any case the changes they made to their version started to conflict with mainstream libfabric 1.15.0 when support for OPX was introduced. They have now fixed that in master and in 1.15 branch but it may take some time before sites get updated, not sure on the timeline but most likely a few months.
If you prefer to downgrade to 1.14.0 or 1.14.1, I'm also fine with that. We just need to make sure that the rpms that get generated remain compatible with 1.14 for now. It's difficult to handle something like that cleanly I think because of that conflicting change they had introduced...

Reply:
Yeah, this is the danger of applying patches and putting the result into production before your patches are accepted upstream.  Had they waited for upstream acceptance, there would have been no conflict.

I do think I would prefer that we "erase" our update of libfabric to 1.15 rather than pinning mercury so tightly.  My fear is that with so many releases in motion that a mercury with the tight pinning will get out into the wild and that that might come back to haunt us in the future with a need to have multiple release streams for mercury also.  We simply don't have the capacity to start trying to maintain multiple release streams of dependencies.

So we need to do two things...
1. Revert the update in the libfabric repo back to something that will work for everyone (1.14.0 or 1.14.1 as you suggested)
2. Remove the 1.15.* libfabric from our repos.

I can do the latter when the former is ready to land.

Reply:
OK that sounds good to me. We'll also need to keep in mind that we'll need to start using libfabric 1.15.x when OPX testing comes along.

Reply:
@brianjmurrell speaking about the issue with Johann, we'll actually go with another plan.
1. I'll patch the libfabric repo to add support for the CXI enum value that's currently missing and creating conflicts with libfabric 1.15.0.
2. We'll just update that spec file to keep using >= 1.14.0

Reply:
Once https://github.com/daos-stack/libfabric/pull/63 is merged, I'll update the spec file.